llvm/lib/IR/ConstantFold.cpp

```diff
@@ -902,11 +902,6 @@ Constant *llvm::ConstantFoldBinaryInstruction(unsigned Opcode, Constant *C1,
         Constant *ExtractIdx = ConstantInt::get(Ty, i);
         Constant *LHS = ConstantExpr::getExtractElement(C1, ExtractIdx);
         Constant *RHS = ConstantExpr::getExtractElement(C2, ExtractIdx);
-
-        // If any element of a divisor vector is zero, the whole op is poison.
-        if (Instruction::isIntDivRem(Opcode) && RHS->isNullValue())
-          return PoisonValue::get(VTy);
```

Comment:
Could you just do `Res = PoisonValue::get(RHS->getType())` and make a poison element?

Reply:
Isn't it handled by the scalar path? See test `vec_select_udiv_poison`. `udiv <i32 42, i32 -7>, <i32 0, i32 1>` is folded into `<i32 poison, i32 -7>`.